pkg/controller/hibernation/hibernation_actuator.go

```diff
@@ -28,3 +28,12 @@ type HibernationActuator interface {
 	// that have not stopped.
 	MachinesStopped(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) (bool, []string, error)
 }
+
+// HibernationActuator is the interface that the hibernation controller
+// for preemptible instances on cloud providers.
+type HibernationPreemptibleMachines interface {
+	// ReplaceMachines uses the remote client to replace the preemptible machines
+	// belonging to the given ClusterDeployment. Since it uses the remote client
+	// it should only be called when the API is reachable.
```

Comment:
What is the significance of the `replaced` return value? From reading the code, it returns `true` if *any* machine was replaced. Is that the intent (vs. if *all* relevant machines were replaced)?

Reply:
at least one was replaced.

Reply:
Would you mind adding that to the docstring?